Sources/SwiftSyntaxBuilderGeneration/Templates/BuildableCollectionNodesFile.swift

```diff
@@ -0,0 +1,257 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2022 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import Foundation
+import SwiftSyntax
+import SwiftSyntaxBuilder
+
+let buildableCollectionNodesFile = SourceFile {
+  ImportDecl(
+    leadingTrivia: .docLineComment(copyrightHeader),
+    path: "SwiftSyntax"
+  )
+
+  for node in SYNTAX_NODES where node.isSyntaxCollection {
+    let type = node.type
+    let elementType = node.collectionElementType
+    let conformances = ["ExpressibleByArrayLiteral", "SyntaxBuildable", type.expressibleAs]
+    
+    StructDecl(
+      leadingTrivia: node.documentation.isEmpty
+        ? []
+        : .docLineComment("/// \(node.documentation)") + .newline,
+      modifiers: [TokenSyntax.public],
+      identifier: type.buildable,
+      inheritanceClause: createTypeInheritanceClause(conformances: conformances)
+    ) {
+      VariableDecl(.let, name: "elements", type: ArrayType(elementType: elementType.buildable))
+
+      InitializerDecl(
+        leadingTrivia: [
+          "/// Creates a `\(type.buildable)` with the provided list of elements.",
+          "/// - Parameters:",
+          "/// - elements: A list of `\(elementType.expressibleAs)`",
+        ].map { .docLineComment($0) + .newline }.reduce([], +),
+        modifiers: [TokenSyntax.public],
+        parameters: ParameterClause(
+          parameterList: [
+            FunctionParameter(
+              firstName: .wildcard,
+              secondName: .identifier("elements"),
+              colon: .colon,
+              type: ArrayType(elementType: elementType.expressibleAs)
+            )
+          ]
+        )
+      ) {
+        SequenceExpr(elements: ExprList([
+          MemberAccessExpr(base: "self", name: "elements"),
+          AssignmentExpr(),
+          elementType.isToken
+            ? "elements"
+            : FunctionCallExpr(MemberAccessExpr(base: "elements", name: "map"), trailingClosure: ClosureExpr {
+                FunctionCallExpr(MemberAccessExpr(base: "$0", name: "create\(elementType.buildableBaseName)"))
+              }),
+        ]))
```

Comment:
The `Array: ExpressibleAsExprList` conformance (as proposed in https://github.com/apple/swift-syntax/pull/465#discussion_r899162231) would really come in handy since we use quite a few `SequenceExpr`s here and in the other generated files.

Perhaps noteworthy that `BuildableCollectionNodes` (this file) is also responsible for generating these conformances, though I would still consider this to be more of a candidate for a future PR.

Reply:
Moving the discussion to #528.